match2/commons/match_group_display_bracket.lua

```diff
@@ -1,738 +1,745 @@
+                or #match.opponents % 2 == 1
+                    and opponentIx == math.floor(#match.opponents / 2) + 1
+        else
+            singleStraightLine = false
         end
-    end
-    
-    out:css('min-height', height .. 'px')
-
-    return out
-end
 
-function _getBracketConfig(args)
-    local globalConfig = Json.parse(tostring(mw.message.new('BracketConfig')))
-    local config = {}
-    for param, format in pairs(BRACKET_CONFIG_FORMAT) do
-        local val = utils.misc.emptyOr(args[param], globalConfig[param], BRACKET_CONFIG_DEFAULT[param])
-        if val ~= nil then
-            if format == 'number' then
-                config[param] = tonumber(val)
-            elseif format == 'boolean' then
-                config[param] = utils.misc.readBool(val)
-            elseif format == 'string' then
-                config[param] = tostring(val)
+        -- Don't show the header if it's disabled. Also don't show the header 
+        -- if it is the first match of a round and if a higher round match can
+        -- show it instead.
+        local isFirstChild = upperMatchIds[matchId] and matchId == matchesById[upperMatchIds[matchId]].bracketData.lowerMatches[1].matchId
+        local showHeader = match.bracketData.header
+            and not config.hideRoundTitles 
+            and not isFirstChild
+
+        local headerFullHeight = showHeader
+            and config.headerMargin + config.headerHeight + math.max(config.headerMargin - config.matchMargin, 0)
+            or 0
+        local matchHeight = #match.opponents * config.opponentHeight
+
+        -- Align the match with its lower round matches
+        local matchTop
+        if singleStraightLine then
+            -- Single straight line: Align the connecting line with the middle 
+            -- of the opponent it connects into.
+            local opponentIx = match.bracketData.lowerMatches[1].opponentIx
+
+            matchTop = lowerLayouts[1].mid 
+                - ((opponentIx - 1) + 0.5) * config.opponentHeight
+
+        elseif 0 < #lowerLayouts then 
+            if #lowerLayouts % 2 == 0 then
+                -- Even number of lower round matches: Align this match to the 
+                -- midpoint of the middle two lower round matches.
+                
+                local aMid = heightSums[#lowerLayouts / 2] + lowerLayouts[#lowerLayouts / 2].mid
+                local bMid = heightSums[#lowerLayouts / 2 + 1] + lowerLayouts[#lowerLayouts / 2 + 1].mid
+                matchTop = (aMid + bMid) / 2 - matchHeight / 2
+
+            else
+                -- Odd number of lower round matches: Align this match to the 
+                -- middle one.
+                local middleLowerLayout = lowerLayouts[math.floor(#lowerLayouts / 2) + 1]
+                matchTop = heightSums[math.floor(#lowerLayouts / 2) + 1] + middleLowerLayout.mid
+                    - matchHeight / 2
```

Comment:
Could consider using functions for both of these and dropping the comments (since the function names should be enough)

Reply:
Do you mean using eta expansion to define a function and run it right away? Or move this to a separate function elsewhere? I'm not sure if moving this code out is a good idea - it makes the code flow jump around too much. I think it's important to compare the odd case with even case, so it's nice to keep them together.

What I mean by eta expansion:
```
matchTop = (function alignWithOddLowerRoundMatchCount()
    -- Align this match to the middle one.
    local middleLowerLayout = lowerLayouts[math.floor(#lowerLayouts / 2) + 1]
    return heightSums[math.floor(#lowerLayouts / 2) + 1] + middleLowerLayout.mid
        - matchHeight / 2
end)()
```